optuna/pruners/threshold.py

```diff
@@ -0,0 +1,59 @@
+from optuna.pruners import BasePruner
+
+from optuna.type_checking import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    from optuna.structs import FrozenTrial
+    from optuna.study import Study
+
+
+class ThresholdPruner(BasePruner):
+    """Pruner to detect abnormal metrics of the trials.
+
+    Prune if a metric exceeds upper bound threshold or
+    falls behind lower bound threshold which users specify.
+
+    Example:
+
+        .. code::
+
+            >>> from optuna import create_study
+            >>> from optuna.pruners import ThresholdPruner
+            >>>
+            >>> study = create_study(pruner=ThresholdPruner(upper=500))
+            >>> study.optimize(objective)
+
+    Args
+        lower_bound:
+            minimum value which determines whether pruner prunes or not
+            (If value is smaller than lower_bound, it prunes)
+        upper_bound:
+            maximum value which determines whether pruner prunes or not
+            (If value is larger than upper_bound, it prunes)
+
+    """
+
+    def __init__(self, lower_bound=None, upper_bound=None):
```

Comment:
I'm wondering if we shouldn't provide a `n_warmup_steps` (https://github.com/optuna/optuna/blob/master/optuna/pruners/percentile.py#L127-L133) as users might want to give the model a few iterations before assessing it's performance. The user could check this him-/herself but this warmup parameter would be consistent with the median and percentile pruners.

Reply:
8199eef
22dcb01
7c6944c

I introduced three members `n_startup_trials`, `n_warmup_steps` and `interval_steps`.